forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/listeners/TaskAssignmentListener.java

```diff
@@ -16,77 +25,180 @@
+    }
+
+    public Message createMessage(InternetAddress[] recipients, String body, String subject, String taskId, Boolean attachPdf, Session session) throws Exception {
         MailConfiguration configuration = getConfiguration();
         Message message = new MimeMessage(session);
         message.setFrom(new InternetAddress(configuration.getSender(), configuration.getSenderAlias()));
-        message.setRecipients(RecipientType.TO, InternetAddress.parse(recipient));
+        Address[] addresses;
+        message.setRecipients(RecipientType.TO, recipients);
 
-        message.setSentDate(new Date());
-        message.setSubject(subject);
+        // Create a multipart message
+        Multipart multipart = new MimeMultipart();
 
+        // Create the body text of the email
         String bodyUrl = body.replace("$BASE_URL", baseUrl);
         String finalBody = bodyUrl.replace("$TASK_ID", taskId);
-
+        BodyPart messageBodyPart = new MimeBodyPart();
         if (!body.isEmpty()) {
-            message.setText(finalBody);
+            messageBodyPart.setText(finalBody);
         } else {
             message.setText("");
         }
+        multipart.addBodyPart(messageBodyPart);
+
+        // Create the pdf attachment if attach is true
+        if (attachPdf) {
+            messageBodyPart = new MimeBodyPart();
+            String filename = "form.pdf";
+            DataSource source = new FileDataSource(filename);
+            messageBodyPart.setDataHandler(new DataHandler(source));
+            messageBodyPart.setFileName(filename);
+            multipart.addBodyPart(messageBodyPart);
+        }
+
+        message.setSentDate(new Date());
+        message.setContent(multipart);
+        message.setSubject(subject);
 
         return message;
     }
 
+    private Template getTemplate() throws IOException {
+        // Instantiate the FreekMarker config
+        Configuration cfg = new Configuration(Configuration.VERSION_2_3_29);
+        cfg.setClassForTemplateLoading(Application.class, "/templates");
+        cfg.setDefaultEncoding("UTF-8");
+        cfg.setTemplateExceptionHandler(TemplateExceptionHandler.RETHROW_HANDLER);
+        cfg.setLogTemplateExceptions(false);
+        cfg.setWrapUncheckedExceptions(true);
+        cfg.setFallbackOnNullLoopVariable(false);
+        Template template = cfg.getTemplate("/form.html");
+        return template;
+    }
+
+    private void RenderPage(File file) {
+        try (Playwright playwright = Playwright.create()) {
+
+            Browser browser = playwright.chromium().launch();
+            Page page = browser.newPage();
+
+            page.navigate(file.toPath().toUri().toURL().toString());
+            page.waitForLoadState(LoadState.DOMCONTENTLOADED);
+
+            page.pdf(new Page.PdfOptions().setPath(Paths.get("form.pdf")).setMargin(new Margin().setRight("40px").setLeft("40px").setTop("40px").setBottom("40px")));
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Similar to my comment above, we should try to handle this one. Maybe add in a `throws Exception` to the method signature, and let it be caught by the exception handler at the top level